sources/accessible-name-and-description.ts

```diff
@@ -348,18 +351,37 @@ export function computeTextAlternative(
 		// window.getComputedStyle(elementFromAnotherWindow) or if I don't bind it
 		// the type declarations don't require a `this`
 		// eslint-disable-next-line no-restricted-properties
-		getComputedStyle = window.getComputedStyle.bind(window),
+		getComputedStyle: _getComputedStyle = window.getComputedStyle.bind(window),
 		hidden = false,
 	} = options;
+	const cachingGetComputedStyle: GetComputedStyle = (
+		el,
+		pseudoElement,
+	): CSSStyleDeclaration => {
+		// we don't cache the pseudoElement styles
+		if (pseudoElement !== undefined) {
+			return _getComputedStyle(el, pseudoElement);
+		}
+		const cachedStyles = computedStyles.get(el);
```

Comment:
Worth calling out that this lookup is not the expected O(1) but O(N) which is especially important since this may negate any benefit we get for caching.

I feel like we should only cache when the built-in `Map` is available. If you're testing in IE11, you don't need the caching anyway since you're in a browser already which has less issues with computing styles.

Reply:
that makes sense. i left a comment in the `getComputedStyles` function about the polyfill being slower. let me know if you want me to document that somewhere else as well

Reply:
I'm saying we shouldn't use the polyfill at all and only cache when we know the lookup is O(1).

Reply:
sorry, i should have been clearer. i left a comment saying it was slower and that was why we were skipping using the caching if `Map` is undefined

Reply:
i just removed the polyfill completely so there won't be a chance someone stumbles into the slow behavior

Reply:
i did end up having to include the global type for Map because otherwise TS complained about needing to change the target